tests/unit/test_hnsw_tiered.cpp

```diff
@@ -3535,3 +3535,64 @@ TYPED_TEST(HNSWTieredIndexTestBasic, FitMemoryTest) {
     // allocation.
     ASSERT_EQ(index->getAllocationSize(), initial_memory) << "fitMemory() after adding 1 vector";
 }
+
+TYPED_TEST(HNSWTieredIndexTest, deleteBothAsyncAndInplace) {
+    // Create TieredHNSW index instance with a mock queue.
+    size_t dim = 4;
+    bool isMulti = TypeParam::isMulti();
+
+    HNSWParams params = {.type = TypeParam::get_index_type(),
+                         .dim = dim,
+                         .metric = VecSimMetric_L2,
+                         .multi = isMulti};
+    VecSimParams hnsw_params = CreateParams(params);
+
+    auto mock_thread_pool = tieredIndexMock();
+
+    auto *tiered_index = this->CreateTieredHNSWIndex(hnsw_params, mock_thread_pool);
+    auto allocator = tiered_index->getAllocator();
+
+    // Insert one vector to HNSW.
+    GenerateAndAddVector<TEST_DATA_T>(tiered_index->backendIndex, dim, 0);
+    // Add another vector and remove it. Expect that at HNSW index one repair job will be created.
+    GenerateAndAddVector<TEST_DATA_T>(tiered_index->backendIndex, dim, 1, 1);
+    ASSERT_EQ(tiered_index->deleteLabelFromHNSW(1), 1);
+    ASSERT_EQ(mock_thread_pool.jobQ.size(), 1);
+
+    // The first job should be a repair job of the first inserted node id (0) in level 0.
+    ASSERT_EQ(mock_thread_pool.jobQ.size(), 1);
+    ASSERT_EQ(mock_thread_pool.jobQ.front().job->jobType, HNSW_REPAIR_NODE_CONNECTIONS_JOB);
+    ASSERT_TRUE(mock_thread_pool.jobQ.front().job->isValid);
+    ASSERT_EQ(((HNSWRepairJob *)(mock_thread_pool.jobQ.front().job))->node_id, 0);
+    ASSERT_EQ(((HNSWRepairJob *)(mock_thread_pool.jobQ.front().job))->level, 0);
+    ASSERT_EQ(tiered_index->idToRepairJobs.size(), 1);
+    ASSERT_GE(tiered_index->idToRepairJobs.at(0).size(), 1);
+    ASSERT_EQ(tiered_index->idToRepairJobs.at(0)[0]->associatedSwapJobs.size(), 1);
+    ASSERT_EQ(tiered_index->idToRepairJobs.at(0)[0]->associatedSwapJobs[0]->deleted_id, 1);
+
+    // Add one more vector and remove it, expect that the same repair job for 0 would be created
+    // for reapring 0->2.
+    GenerateAndAddVector<TEST_DATA_T>(tiered_index->backendIndex, dim, 2, 2);
```

Comment:
```suggestion
    // for repairing 0->2.
```

Reply:
what?

Reply:
😓 got messed up
just fixed the Spelling of repairing